projects/OG-Analytics/src/test/java/com/opengamma/analytics/financial/interestrate/future/provider/SwapFuturesPriceDeliverableTransactionDiscountingMethodE2ETest.java

```diff
@@ -0,0 +1,434 @@
+  private static final double NOTIONAL = 1000.0;
+  private static final NotionalProvider NOTIONAL_PROV = new NotionalProvider() {
+    @Override
+    public double getAmount(final LocalDate date) {
+      return 1.0;
+    }
+  };
+
+  /* EUR */
+  private static final SwapFuturesPriceDeliverableTransaction TRANSACTION_EUR;
+  static {
+    boolean payer = true;
+    LocalDate startDate = LocalDate.of(2014, 6, 18);
+    LocalDate endDate = LocalDate.of(2019, 6, 18);
+    DayCount fixedDc = DayCounts.THIRTY_360;
+    AdjustedDateParameters accrualPeriodParameters = new AdjustedDateParameters(CALENDAR_EUR,
+        INDEX_EUR.getBusinessDayConvention());
+    OffsetAdjustedDateParameters paymentDateParameters = new OffsetAdjustedDateParameters(0, OffsetType.BUSINESS,
+        CALENDAR_EUR, INDEX_EUR.getBusinessDayConvention());
+    OffsetAdjustedDateParameters resetDateParameters = new OffsetAdjustedDateParameters(0, OffsetType.BUSINESS,
+        CALENDAR_EUR, INDEX_EUR.getBusinessDayConvention());
+    OffsetAdjustedDateParameters fixingDateParameters = new OffsetAdjustedDateParameters(-2, OffsetType.BUSINESS,
+        CALENDAR_EUR, BusinessDayConventions.PRECEDING);
+    double rate = 0.015;
+    AnnuityDefinition<?> fixedLeg = new FixedAnnuityDefinitionBuilder().payer(!payer).currency(EUR)
+        .notional(NOTIONAL_PROV).startDate(startDate).endDate(endDate).dayCount(fixedDc)
+        .rollDateAdjuster(RollConvention.NONE.getRollDateAdjuster(0)).accrualPeriodFrequency(Period.ofYears(1)).
+        accrualPeriodParameters(accrualPeriodParameters).paymentDateRelativeTo(DateRelativeTo.END).
+        paymentDateAdjustmentParameters(paymentDateParameters).rate(rate).build();
+    AnnuityDefinition<? extends PaymentDefinition> iborLeg = new FloatingAnnuityDefinitionBuilder().payer(payer).
+        currency(EUR).notional(NOTIONAL_PROV).startDate(startDate).endDate(endDate).dayCount(INDEX_EUR.getDayCount())
+        .rollDateAdjuster(RollConvention.NONE.getRollDateAdjuster(0)).accrualPeriodFrequency(Period.ofMonths(6)).
+        accrualPeriodParameters(accrualPeriodParameters).paymentDateRelativeTo(DateRelativeTo.END).
+        paymentDateAdjustmentParameters(paymentDateParameters).index(INDEX_EUR)
+        .resetDateAdjustmentParameters(resetDateParameters).resetRelativeTo(DateRelativeTo.START).
+        fixingDateAdjustmentParameters(fixingDateParameters).build();
+    SwapFixedIborDefinition swapDefinition = new SwapFixedIborDefinition(toFixedLeg(fixedLeg), toIborLeg(iborLeg));
+    ZonedDateTime lastTradingDate = DateUtils.getUTCDate(2014, 6, 16, 0, 0);
+    SwapFuturesPriceDeliverableSecurityDefinition underlyingSwapFuture = new SwapFuturesPriceDeliverableSecurityDefinition(
+        lastTradingDate, swapDefinition, NOTIONAL);
+    SwapFuturesPriceDeliverableTransactionDefinition SwapFutureTransaction = new SwapFuturesPriceDeliverableTransactionDefinition(
```

Comment:
use lower case first letter for variables: swapFuture... Also in other transaction definitions.